plugins/setoid_ring/newring.ml

```diff
@@ -146,17 +146,21 @@ let ic_unsafe c = (*FIXME remove *)
   let sigma = Evd.from_env env in
   fst (Constrintern.interp_constr env sigma c)
 
-let decl_constant na univs c =
+let decl_constant name univs c =
   let open Constr in
   let vars = CVars.universes_of_constr c in
   let univs = UState.restrict_universe_context ~lbound:(Global.universes_lbound ()) univs vars in
   let () = DeclareUctx.declare_universe_context ~poly:false univs in
   let types = (Typeops.infer (Global.env ()) c).uj_type in
   let univs = Monomorphic_entry Univ.ContextSet.empty in
-  mkConst(declare_constant ~name:(Id.of_string na)
+  mkConst(declare_constant ~name
             ~kind:Decls.(IsProof Lemma)
             (DefinitionEntry (definition_entry ~opaque:true ~types ~univs c)))
 
+let decl_constant na suff univs c =
```

Comment:
Is ring the right place for this? Is there no other code needing to define a fresh constant with a suffix?

Reply:
We're using the ring-specific decl_constant which makes some non general choices (have a look at it) so it would just be noisy to move it out imo

Reply:
Ah ok, I missed that.